Neuro/Tasks/TasksHandler.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections;
+using Neuro.DependencyInjection;
+using Neuro.Utilities;
+using Reactor.Utilities.Attributes;
+using UnityEngine;
+
+namespace Neuro.Tasks;
+
+[RegisterInIl2Cpp]
+public class TasksHandler : MonoBehaviour, ITasksHandler
+{
+    public TasksHandler(IntPtr ptr) : base(ptr) { }
+
+    public IContextProvider Context { get; set; }
```

Comment:
i'd recommend using {get; init;} for things you don't want to set after the object is constructed. init-properties work with DI containers; the init thing is really a set; with a compile-time attribute to let the compiler know to not allow certain types of statements. the reflection a DI container does is able to set these properties.

Reply:
In this case we cannot use init for objects which are MonoBehaviours because they cannot assign anything in the constructor, only after Awake. 

Regardless I removed DI in my latest commit.

Reply:
init; only properties allow you to set the property *once* after construction using object-initializer syntax. it's strictly a compile-time concept to allow you to do: new thing() { InitOnlyProperty = 5 }. It does not affect the runtime semantics at all (vs. a set;) DI is a very useful thing.

Reply:
Yeah I know, the init keyword is implemented as a set with some special attributes. However our object creation is GameObject.AddComponent, which is not an object initializer. (I said constructor in my previous messsage my bad, I meant initializer)

Reply:
I removed DI after a conversation with @js6pak on discord, where we realized that it's overkill and would most likely go unused, since we are in a unity environment and cannot run unit tests, nor will we want to programmatically replace the instance in any way